jujugui/static/gui/src/app/components/account/credentials/credentials.js

```diff
@@ -0,0 +1,218 @@
+      getCloudCredentialNames: React.PropTypes.func.isRequired,
+      getCloudProviderDetails: React.PropTypes.func.isRequired,
+      listClouds: React.PropTypes.func.isRequired,
+      revokeCloudCredential: React.PropTypes.func.isRequired,
+      username: React.PropTypes.string.isRequired
+    },
+
+    getInitialState: function() {
+      this.xhrs = [];
+      return {
+        credentials: [],
+        loading: false
+      };
+    },
+
+    componentWillMount: function() {
+      this._getClouds();
+    },
+
+    componentWillUnmount: function() {
+      this.xhrs.forEach((xhr) => {
+        xhr && xhr.abort && xhr.abort();
+      });
+    },
+
+    /**
+      Retrieve the list of clouds.
+
+      @method _getClouds
+    */
+    _getClouds: function() {
+      const xhr = this.props.listClouds((error, clouds) => {
+        if (error) {
+          const message = 'Unable to list clouds';
+          this.props.addNotification({
+            title: message,
+            message: `${message}: ${error}`,
+            level: 'error'
+          });
+          console.error(message, error);
+          return;
+        }
+        this._getCloudCredentialNames(clouds);
+      });
+      this.xhrs.push(xhr);
+    },
+
+    /**
+      Retrieve the list of credential names for the user.
+
+      @method _getCloudCredentialNames
+      @param {Array} clouds A list of cloud ids.
+    */
+    _getCloudCredentialNames: function(clouds) {
+      const pairs = Object.keys(clouds).map(cloud => {
+        return [this.props.username, cloud];
+      });
+      this.setState({loading: true}, () => {
+        const xhr = this.props.getCloudCredentialNames(
+          pairs, (error, names) => {
+            if (error) {
+              const message = 'Unable to get names for credentials';
+              this.props.addNotification({
+                title: message,
+                message: `${message}: ${error}`,
+                level: 'error'
+              });
+              console.error(message, error);
+              return;
+            }
+            let credentials = [];
+            names.forEach((cloud, i) => {
+              cloud.names.forEach(name => {
+                credentials.push({
+                  name: name,
+                  // Store the cloud for this name.
+                  cloud: pairs[i][1]
+                });
+              });
+            });
+            this.setState({
+              credentials: credentials,
+              loading: false,
+            });
+          });
+        this.xhrs.push(xhr);
+      });
+    },
+
+    /**
+      Handle deleting a credential.
+
+      @method _handleDeleteCredential
+      @param credential {String} A credential id.
+    */
+    _handleDeleteCredential: function(credential) {
+      const xhr = this.props.revokeCloudCredential(credential, (error) => {
+        if (error) {
+          const message = 'Unable to revoke the cloud credential';
+          this.props.addNotification({
```

Comment:
Thanks for pulling in addNotification and doing something more with the error than just logging to the console.